modules/edge-governance/src/governance.rs

```diff
@@ -106,6 +111,9 @@ decl_module! {
 			let hash = T::Hashing::hash(&buf[..]);
 			ensure!(<ProposalOf<T>>::get(hash) == None, "Proposal already exists");
 
+			// Burn the proposal creation bond amount
+			ensure!(T::Currency::can_slash(&_sender, Self::proposal_creation_bond()), "Not enough currency for slashing bond");
+			T::Currency::slash(&_sender, Self::proposal_creation_bond());
```

Comment:
As we discussed in the meeting, I think it makes more sense to do a reserve rather than a slash, as we'll ultimately be returning the funds. This is how they tend to do it in substrate: https://github.com/paritytech/substrate/blob/master/srml/council/src/seats.rs#L202

Reply:
But we won't be returning the funds if the identities are denied?

Reply:
It's as if we're charging you to participate, but then rewarding you if you participated correctly. I'm not so sure reserving the balance acts the same.

Reply:
The way it works is that you reserve the funds, return them if the identity is accepted, or slash them if the identity is denied. I'm pretty confident this is what the currency reserve functionality was designed for. It's effectively a temporary bond, but you lose it if you misbehave.

I expect that if we choose to instead manipulate the entire currency supply, weird things will happen, as various functions are pegged to changes in the overall supply (e.g. the treasury pot will automatically scale to be the same proportion of the overall currency supply).